Sources/Workspace/Workspace.swift

```diff
@@ -40,7 +40,15 @@ public enum WorkspaceError: Swift.Error {
     case incompatibleToolsVersion(rootPackage: AbsolutePath, required: ToolsVersion, current: ToolsVersion)
 
     /// The package at edit destination is not the one user is trying to edit.
-    case mismatchingDestinationPackage(path: AbsolutePath, destPackage: String, expectedPackage: String)
+    case mismatchingDestinationPackage(path: AbsolutePath, destinationPackage: String, expectedPackage: String)
+    
+    /// The dependency can't be pinned because it is being edited.
+    case cantPinDependencyInEditMode(dependency: ManagedDependency)
+}
+
+/// A warning in one of the workspace operations.
+public enum WorkspaceWarning {
```

Comment:
It would be nice to declare this in Workspace class as "Warning" and then use extension to define diagnostics in a sep class (to avoid clutter).